From: David Bauer <mail@david-bauer.net>
Date: Wed, 17 Jun 2026 13:42:31 +0200
Subject: wifi: mt76: mt76x2: receive BlockAck frames

Enable RX of BlockAck frames to the host.

This is required to track delivery state of BlockAckReq frames.

Signed-off-by: David Bauer <mail@david-bauer.net>

diff --git a/package/kernel/mt76/patches/0506-wifi-mt76-mt76x2-receive-BlockAck-frames.patch b/package/kernel/mt76/patches/0506-wifi-mt76-mt76x2-receive-BlockAck-frames.patch
new file mode 100644
index 0000000000000000000000000000000000000000..1bef25c4794a00adbcf60b1ba0bcb51aa95d6f7c
--- /dev/null
+++ b/package/kernel/mt76/patches/0506-wifi-mt76-mt76x2-receive-BlockAck-frames.patch
@@ -0,0 +1,37 @@
+From 7e84d0d8660d7cbfa5cadf94ffa709993e387877 Mon Sep 17 00:00:00 2001
+From: David Bauer <mail@david-bauer.net>
+Date: Tue, 16 Jun 2026 19:07:23 +0200
+Subject: [PATCH 1/2] wifi: mt76: mt76x2: receive BlockAck frames
+
+Enable RX of BlockAck frames to the host.
+
+This is required to track delivery state of BlockAckReq frames.
+
+Signed-off-by: David Bauer <mail@david-bauer.net>
+---
+ mt76x02_util.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/mt76x02_util.c b/mt76x02_util.c
+index e5d9d1bc..2b9e843a 100644
+--- a/mt76x02_util.c
++++ b/mt76x02_util.c
+@@ -217,6 +217,7 @@ void mt76x02_configure_filter(struct ieee80211_hw *hw,
+ 	mutex_lock(&dev->mt76.mutex);
+ 
+ 	dev->mt76.rxfilter &= ~MT_RX_FILTR_CFG_OTHER_BSS;
++	dev->mt76.rxfilter &= ~MT_RX_FILTR_CFG_BA;
+ 
+ 	MT76_FILTER(FCSFAIL, MT_RX_FILTR_CFG_CRC_ERR);
+ 	MT76_FILTER(PLCPFAIL, MT_RX_FILTR_CFG_PHY_ERR);
+@@ -224,7 +225,6 @@ void mt76x02_configure_filter(struct ieee80211_hw *hw,
+ 			     MT_RX_FILTR_CFG_CTS |
+ 			     MT_RX_FILTR_CFG_CFEND |
+ 			     MT_RX_FILTR_CFG_CFACK |
+-			     MT_RX_FILTR_CFG_BA |
+ 			     MT_RX_FILTR_CFG_CTRL_RSV);
+ 	MT76_FILTER(PSPOLL, MT_RX_FILTR_CFG_PSPOLL);
+ 
+-- 
+2.53.0
+
